include/boost/numeric/ublas/tensor/detail/extents_functions.hpp

```diff
@@ -160,7 +159,7 @@ template <class ExtentsType>
 inline 
 constexpr bool valid(ExtentsType const &e) {
   
-  static_assert(detail::is_extents<ExtentsType>::value, "boost::numeric::ublas::valid() : invalid type, type should be an extents");
+  static_assert(is_extents_v<ExtentsType>, "boost::numeric::ublas::valid() : invalid type, type should be an extents");
 
   using size_type = typename ExtentsType::size_type;
   using value_type = typename ExtentsType::value_type;
```

Comment:
I recommend to use the following code:
```cpp
if   ( e.size() ==1u) return e[0] == 1u;
else ( e.size() > 1u) return std::all_of(e.begin(), e.end(), [](auto a) { return a > 0u; });
else                  return false;
```
Why because it makes your code more readable. 
using `1u` instead of `size_type(1)` is save as we know that `deltype(e.size())` should be an unsigned type
using `1u` instead of `value_type(1)` is save we know that `deltype(e.front())` should be an unsigned type
etc.

Reply:
This was to support any type extents even if a user creates his own he/she might change the type of size.

Reply:
I can not use 1u because we do not know the type of extents.

Reply:
but you know that (or we should guarantee that)
```cpp
is_unsigned_v<typename extents_traits<extents_t>::size_type> == true
```
and
```cpp
is_unsigned_v<typename extents_traits<extents_t>::value_type> == true
```
So comparing with values with `1u` should always work even if number of bits differ.

Reply:
Changed it.